product_kit_management/models/sale_order_line.py

```diff
@@ -0,0 +1,73 @@
+from odoo import Command, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    is_kit = fields.Boolean(string="Is Kit", related="product_id.is_kit", store=True)
+    kit_component_ids = fields.One2many(
+        'product.kit.component', 'parent_order_line_id',
+        compute="_compute_sub_products", store=True, string="Kit Components",
+        help="List of sub products"
+    )
+    is_kit_component = fields.Boolean(string="Is Kit Component?", default=False)
+    kit_parent_line_id = fields.Many2one(
+        'sale.order.line', string="Parent Kit Line",
+        help="Refrence to the main kit product's sale order line"
+    )
+
+    def open_kit_wizard(self):
+        return{
+            'type': 'ir.actions.act_window',
+            'name': 'Product Kit Wizard',
+            'res_model': 'product.kit',
+            'view_mode': 'form',
+            'target': 'new',
+            'context': {'default_sale_order_line_id': self.id}
+        }
+
+    @api.depends('product_id')
+    def _compute_sub_products(self):
+        """
+        Computes the sub-products (kit components) for a sale order line.
+        It retrieves the sub-products linked to the main product and updates
+        the `kit_component_ids` field accordingly.
+        """
+        for line in self:
+            product_template = line.product_id.product_tmpl_id
+            for sub_product in product_template.sub_product_ids:
+                if sub_product.id not in line.kit_component_ids.mapped('kit_component_id').ids:
+                    line.kit_component_ids = [
+                        Command.create({
+                            'parent_order_line_id': line.id,
+                            'kit_component_id': sub_product.id,
+                            'price': sub_product.lst_price or 0.0,
+                        })
+                    ]
+
+    @api.ondelete(at_uninstall=False)
+    def _delete_sub_products(self):
+        """
+        Ensures that when a kit product is deleted from a sale order,
+        all its linked kit component lines are also removed.
+        """
+        lines_with_kit = self.filtered(lambda l: l.is_kit)
+        if lines_with_kit:
+            self.env['sale.order.line'].search([
+                ('order_id', 'in', lines_with_kit.mapped('order_id').ids),
```

Comment:
```suggestion
                ('order_id', 'in', lines_with_kit.order_id.ids),
```
for relational fields you don't need to use mapped